src/sleep/sleep.c

```diff
@@ -621,6 +598,19 @@ static int run(int argc, char *argv[]) {
                                        "Sleep operation \"%s\" is disabled by configuration, refusing.",
                                        sleep_operation_to_string(arg_operation));
 
+        /* Freeze the user sessions */
+        r = getenv_bool("SYSTEMD_SLEEP_SKIP_FREEZE_USER_SESSIONS");
+        if (r < 0 && r != -ENXIO)
+                log_warning_errno(r, "Cannot parse value of $SYSTEMD_SLEEP_SKIP_FREEZE_USER_SESSIONS, ignoring.");
+        if (r != 1) {
+                r = unit_freezer_freeze(SPECIAL_USER_SLICE, &user_slice_freezer);
+                if (r < 0)
+                        log_warning_errno(r, "Failed to freeze user sessions, ignoring: %m");
+                else
+                        log_info("Froze user sessions");
+        } else
+                log_info("Not freezing user sessions!");
```

Comment:
i think we should raise this to LOG_NOTICE, and give a clearer explanation why this is a bad idea and lowers security.

i mean, we expect that this codepath is hit on nvidia systems, and we want to put some shame on nvidia here to unfuck this, hence this should look "ugly" to some degree and make people afraid.

Hence: "User sessions remain unfrozen on explicit request. This is not recommended, and might create deadlocks, security leaks and undesired behaviour if home directory encryption or hybrid sleeping is used."